approaching_emergency_vehicle_plugin/src/approaching_emergency_vehicle_plugin_node.cpp

```diff
@@ -475,16 +510,40 @@ namespace approaching_emergency_vehicle_plugin
       // ERV cannot be tracked since its route could not be generated; return an empty object
       return boost::optional<ErvInformation>();
     }
-
+    
     // Determine the ERV's current lane index
     // Note: For 'lane index', 0 is rightmost lane, 1 is second rightmost, etc.; Only the current travel direction is considered
     if(!erv_future_route.get().shortestPath().empty()){
       lanelet::ConstLanelet erv_current_lanelet = erv_future_route.get().shortestPath()[0];
+
+      if (is_same_direction_.find(erv_information.vehicle_id) == is_same_direction_.end()) // no need to check again if direction is set already
+      {
+        is_same_direction_[erv_information.vehicle_id] = false;
+        for (auto llt: erv_future_route.get().shortestPath()) // checks if ERV is on the same path assuming CMV got all of its planned route when detected
+        {
+          if (wm_->getRoute()->contains(llt))
+          {
+            is_same_direction_[erv_information.vehicle_id] = true;
+
+            RCLCPP_DEBUG_STREAM(rclcpp::get_logger(logger_name), "Detected that ERV: " << erv_information.vehicle_id << " and CMV are on the SAME direction");
+            break;
+          }
+        }
+      }
+
+      if (!is_same_direction_[erv_information.vehicle_id])  // opposite direction
+      {
+        RCLCPP_DEBUG_STREAM(rclcpp::get_logger(logger_name), "Detected that ERV and CMV are on the DIFFERENT direction");
+        return boost::optional<ErvInformation>(); // if opposite direction, do not track
+      }
+
       erv_information.lane_index = wm_->getMapRoutingGraph()->rights(erv_current_lanelet).size();
+
+     
       RCLCPP_DEBUG_STREAM(rclcpp::get_logger(logger_name), "ERV's lane index is " << erv_information.lane_index);
     }
     else{
-      RCLCPP_DEBUG_STREAM(rclcpp::get_logger(logger_name), "ERV's shortest path is empty!");
+      RCLCPP_ERROR_STREAM(rclcpp::get_logger(logger_name), "ERV's shortest path is empty!");
```

Comment:
Do you want it as an error or its changed for debugging?

Reply:
I think this should probably be WARN message